lib/rucio/db/sqla/migrate_repo/versions/f1b14a8c2ac1_postgres_use_check_constraints.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2013-2019 CERN for the benefit of the ATLAS collaboration.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - Robert Illingworth <illingwo@fnal.gov>, 2019
+
+''' postgres_use_check_constraints '''
+
+from alembic import context
+from alembic.op import (create_foreign_key, drop_constraint, execute)
+
+
+# Alembic revision identifiers
+revision = 'f1b14a8c2ac1'
+down_revision = 'b8caac94d7f0'
+
+
+def upgrade():
+    '''
+    Upgrade the database to this revision
+    '''
+
+    if context.get_context().dialect.name != 'postgresql':
```

Comment:
i'm a friend of negations as much as anyone else, but wouldn't an indented block with `dialect == "postgresql"` be more descriptive? :-D

Reply:
You can see that it only applies to PostgreSQL right at the beginning without having to scroll to the end to look for an else clause (and originally I thought that the whole thing would be even longer than it turned out).

Reply:
it's fine with me ;-)

Reply:
Just to confirm, for every migrate script after this one, we can simply rely on the `create_constraint/drop_constraint` syntax of alembic, correct?

Reply:
I believe so. This migration script should make the constraints consistent, so there will always be something for `drop_constraint` to drop, and the type will be correct for `create_check_constraint`.

Reply:
As a quick test I took the migrate script from #1864, changed the postgresql upgrade to use the  same code block as Oracle with just `drop_constraint/create_check_constraint`, and it applies cleanly.